nncf/openvino/quantization/weights_compression.py

```diff
@@ -46,35 +349,56 @@ def insert_pre_compression_operations(model: ov.Model, bits: int = 8) -> None:
             weight_node = get_operation_const_op(node, const_port_id)
             if weight_node is None:
                 continue
-
+            if weight_node.name in all_quantized_nodes:
+                continue
             weight_output = weight_node.output(0)
             weight_name = weight_node.get_friendly_name()
             target_inputs = weight_output.get_target_inputs()
 
             original_weight_dtype = weight_output.get_element_type().to_dtype()
             if original_weight_dtype not in [np.float32, np.float16, np.float64]:
                 continue
-
-            weight = get_const_value(weight_node)
             axes = _get_reduction_axes(metatype, node, const_port_id)
-            min_values = np.min(weight, axis=axes, keepdims=True)
-            max_values = np.max(weight, axis=axes, keepdims=True)
+            fq_name = f"{node.get_friendly_name()}/fq_weights_{const_port_id}"
+            weight = get_const_value(weight_node)
+            num_weights = weight.size
+            weight_params = WeightNodeParams(axes, num_weights, fq_name, weight_node, original_weight_dtype)
+            all_weight_params.append(weight_params)
+            all_quantized_nodes.append(weight_node.name)
```

Comment:
@alexsu52 do you remember OV models with shared Const node but with different names in xml? Can we rely on `friendly_name` for consistency with other algorithms, or should we use ``name`?

Reply:
Yes, I remember this case. I don't know about `name`, but comparing pointers to data should work.

Reply:
@alexsu52 use address instead of name now